Automattic/wp-calypso
# update(account): remove `formBase` mixin

#### Changes proposed in this Pull Request

Account is the last piece which uses the `formBase` mixin. In this PR we attempt to remove it in preparation for the reduxification of the `<Account />` component which is served under `/me/account`. This PR is partly a spin-off from #49105 and precedes PRs which refactor `<Account />` to a class component and then fully switch to Redux.

* Account: incorporate `formBase` functionality and remove the `formBase` mixin
* 6f4e787 and 7e2a6ce are tangential; the former fixes a bug where appending the query string `?updated=success` wouldn't result in showing a success notice after changing the username (this is currently broken in prod)

#### Testing instructions

* Go to `/me/account`
* Make sure changing & saving settings works exactly like on prod
* Append `?updated=success` to the URL and verify a success notice is shown (this can also be triggered by changing a username on a non-a8c account)

Related to #24162
